From 80c677c62b0de687bddea4bd4f55e3b592166a34 Mon Sep 17 00:00:00 2001
From: Zhen Lei <thunder.leizhen@huawei.com>
Date: Sat, 21 May 2022 12:19:05 +0800
Subject: [PATCH] arm64: Fix some build errors related to KEXEC
MIME-Version: 1.0
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: 80c677c62b0de687bddea4bd4f55e3b592166a34
Modified-by-SEL: No

Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I575ZA
CVE: NA

-------------------------------------------------------------------------

Fix below build errors:
error: implicit declaration of function ‘reserve_crashkernel_high’
undefined reference to `request_quick_kexec_res'
undefined reference to `reserve_quick_kexec'

1. An empty reserve_crashkernel_high() should be defined when
   CONFIG_KEXEC_CORE is not set.
2. CONFIG_ARM64_PMEM_RESERVE needs to be changed to CONFIG_QUICK_KEXEC.
   This is a mistake.

Fixes: d61f4ca086e9 ("arm64: quick_kexec: Move to stand-alone file")
Fixes: baac34dd89ca ("arm64: kdump: Use page-level mapping for the high memory of crashkernel")
Signed-off-by: Zhen Lei <thunder.leizhen@huawei.com>
Reviewed-by: Hanjun Guo <guohanjun@huawei.com>
Reviewed-by: Kefeng Wang <wangkefeng.wang@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 arch/arm64/mm/init.c     | 4 ++++
 arch/arm64/mm/internal.h | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/mm/init.c b/arch/arm64/mm/init.c
index 7cee9bd13017..b19bdd48cc43 100644
--- a/arch/arm64/mm/init.c
+++ b/arch/arm64/mm/init.c
@@ -70,6 +70,10 @@ phys_addr_t arm64_dma_phys_limit __ro_after_init;
 static void __init reserve_crashkernel(void)
 {
 }
+
+static void __init reserve_crashkernel_high(void)
+{
+}
 #endif
 
 /*
diff --git a/arch/arm64/mm/internal.h b/arch/arm64/mm/internal.h
index e1c6fc36b3b5..9b8e20d87172 100644
--- a/arch/arm64/mm/internal.h
+++ b/arch/arm64/mm/internal.h
@@ -14,7 +14,7 @@ static inline void __init setup_reserve_pmem(u64 start, u64 size) {}
 static inline void __init reserve_pmem(void) {}
 static inline void __init request_pmem_res_resource(void) {}
 #endif
-#ifdef CONFIG_ARM64_PMEM_RESERVE
+#ifdef CONFIG_QUICK_KEXEC
 void __init reserve_quick_kexec(void);
 void __init request_quick_kexec_res(struct resource *res);
 #else
-- 
2.34.1

